### DISCUSSION TITLE
Use bean introspection correctly

### DISCUSSION THREAD

**Discussion by nbrugger-tgm** [micronaut-projects/micronaut-core#7702] [Q&A]
Hi, my question evolves arround how micronauts autogeneration & introspecton works.

As far as i understand the micronaut annotation processor generates `$Definition` and `$Definition$Reference` files whos job is to reflect te information that Micronaut needs for injection, interception AND also provide a more eficient way of reflection. I tried to write a test like this
 ```java
@Singleton
public static class SomeController implements SomeApi {
    @Override
    public void addSomething(String theThing) {
        ...
     }
}
@Test
void obtainMethodReference() {
    ApplicationContext ctx = testContext();
    var controller = ctx.getBeanDefinition(SomeController.class);
    var methodImpl = controller.findMethod("addSomething", String.class).orElseThrow();
}
 ```
But `orElseThrow` throws since it doesn't finds the method.

After further investiagation I found that in the `$Definition` and `$Definition$Reference` do not contain any trace of the method. Also i found out that it should be defined in a `$Exec` file which didn't exists. So my first guess is that micronaut only generates things that it needs for CDI and AOP? (When i annotate the method with an AOP advice it is generated)

Am i miss-using the micronaut beancontext introspection? 
The problems arising from this are quite numerous since this completely removes any trace of the annotations present on `SomeApi#addSomething`.

**Comment by graemerocher** [comment]
is there an annotation on the interface method?

**Reply by nbrugger-tgm** [reply]
Not on the method but its parameters are annotated (`@Size(..,.)`), which should make the method candidate for `@Validated` -> Use case #7703

**Comment by dstepanov** [comment]
You need to annotate the method with `@Executable`

**Reply by nbrugger-tgm** [reply]
What does it do? Is it mentioned in the docs/wiki etc. The use case is to separate the annotations which are kind of boilerplate from the implementation (e.g. definition | implementation) like javax.validation
